File: perl/tests/301-show-userfile.pl
my $normal      = &NormalHandle;
our $userfiles_list_300;

die "SKIPPED: no Userfiles\n" if scalar(@$userfiles_list_300) == 0;

my $file_id   = $userfiles_list_300->[0]->{'id'};
my $file_info = $normal->show_userfile($file_id);

if ($DEBUG) {
  my $dump_file_info = Data::Dumper::DumperX($file_info);
  print "DP_INFO: $dump_file_info\n";
}

die "Value returned not an hash.\n" unless defined($file_info) && ref($file_info) eq "HASH";
die "You have no information for your userfile\n" if scalar(%$file_info) eq "0";
die "You have no entry for log"                   unless exists $file_info->{"log"};
die "You have no entry for syncstatus"            unless exists $file_info->{"remote_sync_status"};
die "You have no entry for parent_id"             unless exists $file_info->{"parent_id"};
die "You have no entry for children_ids"          unless exists $file_info->{"children_ids"};


1;

File: ruby/tests/301-show-userfile.rb
normal   = normal_handle

raise "SKIPPED: no Userfiles" unless @userfiles_list_300.present?

file_id   = @userfiles_list_300.first[:id]
file_info = normal.show_userfile(file_id)

puts "USERFILE_INFO: #{file_info.inspect}" if @debug

raise "Value returned is not an hash" unless file_info.present? && file_info.is_a?(Hash)
raise "You have no information for your userfile" if file_info.empty?
raise "You have no entry for log"                 unless file_info.has_key?(:log)
raise "You have no entry for syncstatus"          unless file_info.has_key?(:remote_sync_status)
raise "You have no entry for parent_id"           unless file_info.has_key?(:parent_id)
raise "You have no entry for children_ids"        unless file_info.has_key?(:children_ids)

true
